### PR TITLE
Integration of create vendor profile

### PR DESCRIPTION
**Pull request by RevathyRajeevan**

### CODE REVIEW

File: src/app/features/vendor/services/vendor.service.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { IMarket } from '../models/vendor.model';
+import { IService } from '../models/vendor.model';
+import { IVendorCreation } from '../models/vendor.model';
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/services/vendor.service.ts]: consolidate into a single import statement for line 4,5,6

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { Component } from '@angular/core';
 import { phonePattern } from '../config/vendor-config';
 import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { IMarket } from '../models/vendor.model';
+import { IService } from '../models/vendor.model';
+import { IVendorCreation } from '../models/vendor.model';
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: consolidate into a single import statement

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 3]
```diff
@@ -72,8 +76,35 @@ ngOnInit(){
   });
 }
 
+//function for fetching markets form table
+fetchMarkets() {
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: add return type, update it for all methods without return type

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 4]
```diff
@@ -47,21 +51,21 @@ export class VendorCreationComponent {
...
 states: string[] = ['California', 'Berlin', 'Sydney', 'Rio de Janeiro'];
+markets!: IMarket[];
+services!: IService[];
 
-//just for temporarily fetching market and service names onto dropdown
-markets: string[] = ['North America', 'Europe', 'Asia-Pacific', 'South America', 'Africa'];
-services: string[] = ['Consulting', 'Software Development', 'Customer Support', 'Cloud Services', 'Marketing'];
-
-constructor(private readonly fb: FormBuilder) {}
+constructor(private readonly fb: FormBuilder,private vendorService: VendorService) {}
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: correct the format

File: src/app/features/vendor/services/vendor.service.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+@Injectable({
+  providedIn: 'root',
+})
+export class VendorService {
+
+  private baseUrl = 'http://localhost:5255/api';  
+
+  constructor(private http: HttpClient) {}
+
+  // Fetching markets from the backend
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/services/vendor.service.ts:15]: Follow JSDoc comment format

File: src/app/features/vendor/models/vendor.model.ts [hunk 6]
```diff
@@ -9,4 +9,27 @@ export interface IVendor {
...
+  isApproved: boolean;
+  marketIds: number[];
+}
+
+export interface IService {
+  id: number;
+  name: string;
+}
+
+export interface IMarket {
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/models/vendor.model.ts]: duplicate interface, rename the interface use it for all drop down data

File: src/app/features/vendor/vendor-creation/vendor-creation.component.html [hunk 7]
```diff
@@ -116,10 +192,23 @@ <h2>New vendor</h2>
                     placeholder="Enter Number"
                     [class.invalidInput]="
                       addVendorForm.get('phone')?.invalid &&
-                      addVendorForm.get('phone')?.touched
-                    "
+                      addVendorForm.get('phone')?.touched"
                   />
                 </p-inputGroup>
+                <div
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.html]: use a common template for showing error messages

Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.html]: done

File: src/app/features/vendor/models/vendor.model.ts (hunk 6 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/models/vendor.model.ts]: done

File: src/app/features/vendor/services/vendor.service.ts (hunk 5 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/services/vendor.service.ts:15]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 4 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 3 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 2 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/services/vendor.service.ts (hunk 1 above)
Comment (RevathyRajeevan) [reply src/app/features/vendor/services/vendor.service.ts]: done

**Review by athul-sanjeevan — APPROVED** [review]

File: src/app/features/vendor/services/vendor.service.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    return this.http.get<IDropDownFields[]>(`${this.baseUrl}/Market/GetMarkets`);
+  }
+
+  // Fetching services from the backend
+  getServices(): Observable<IDropDownFields[]> {
+    return this.http.get<IDropDownFields[]>(`${this.baseUrl}/Service/GetService`);
+  }
+
+  // Posting vendor data to the backend
+  addVendor(vendorData: any): Observable<IVendorCreation> {
```
Comment (kavyamolr) [inline src/app/features/vendor/services/vendor.service.ts]: Provide appropriate type instead of any

**Review by kavyamolr — APPROVED** [review]

**Pull request merged**